### PR TITLE
refactor: Remove no-op TIME_INIT on deser

### PR DESCRIPTION
**Pull request by maflcko**
Split out from https://github.com/bitcoin/bitcoin/pull/24697

### CODE REVIEW

File: src/bitcoin-util.cpp [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #include <streams.h>
 #include <util/system.h>
 #include <util/translation.h>
+#include <version.h>
```
Comment (laanwj) [inline src/bitcoin-util.cpp:18]: Looks like an unrelated change snuck in here?

Comment (MarcoFalke) [reply src/bitcoin-util.cpp:18]: No, the changes here wouldn't compile without this change. 

Commit dbcb5742c48fd26f77e500291d7083e12eec741b removed `INIT_PROTO_VERSION` from protocol.h, but didn't remove the version.h include from protocol, so I am doing that here.

Comment (laanwj) [reply src/bitcoin-util.cpp:18]: Thanks, makes sense.

**Review by fanquake — APPROVED** [review]
ACK fa243e93138d899f41be02af3fa2b029dd8d525e
